### PR TITLE
PT-1284 Integrate Tidepool master for shoreline

### PR DESCRIPTION
**Pull request by CedricPerotto**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,33 +1,24 @@
...
-COPY . .
-RUN go get
+RUN adduser -D tidepool && \
+    chown -R tidepool /go/src/github.com/tidepool-org/shoreline
+RUN apk add --no-cache git
+USER tidepool
+COPY --chown=tidepool . .
 RUN ./build.sh
-
 CMD ["./dist/shoreline"]
```
Comment (yvon-dblg) [inline Dockerfile:13]: Is it still a userful line?

File: shoreline.go [hunk 2]
```diff
@@ -164,40 +191,50 @@ func main() {
 	 * User-Api setup
 	 */
 
-	log.Print(shoreline_service_prefix, "adding ", user.USER_API_PREFIX)
+	var marketoManager marketo.Manager
+	if err := config.User.Marketo.Validate(); err != nil {
+		logger.Println("WARNING: Marketo config is invalid", err)
```
Comment (yvon-dblg) [inline shoreline.go:196]: Maybe we do not want to print this warning message every time this service start ?
For us, it should be more an information like: "Marketo is disabled" ?

File: user/api.go [hunk 3]
```diff
@@ -23,6 +22,126 @@ import (
...
+	})
+	statusMissingUsrDetailsCounter = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "statusMissingUsrDetailsCounter",
+		Help: "The total number of STATUS_MISSING_USR_DETAILS errors",
+	})
+	statusErrorUpdatingPwCounter = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "statusErrorUpdatingPwCounter",
+		Help: "The total number of STATUS_ERROR_UPDATING_PW errors",
+	})
+	statusMissingIdPwCounter = promauto.NewCounter(prometheus.CounterOpts{
```
Comment (yvon-dblg) [inline user/api.go:81]: I have a lint warning here:
```
var statusMissingIdPwCounter should be statusMissingIDPwCounter
```

File: user/api.go [hunk 4]
```diff
@@ -828,7 +961,7 @@ func (a *Api) RefreshSession(res http.ResponseWriter, req *http.Request) {
 
 	if td.IsServer == false && td.DurationSecs > two_hours_in_secs {
```
Comment (yvon-dblg) [inline user/api.go]: This if do not seems useful anymore.
like the ugly var name `two_hours_in_secs`

**Review by yvon-dblg — COMMENTED** [review]
Few questions here & there.

File: Dockerfile (hunk 1 above)
Comment (CedricPerotto) [reply Dockerfile:13]: Which one ?

File: shoreline.go (hunk 2 above)
Comment (CedricPerotto) [reply shoreline.go:196]: For me an absent configuration is indeed invalid.

File: user/api.go (hunk 3 above)
Comment (CedricPerotto) [reply user/api.go:81]: The whole shoreline project is a big bag of lint errors. 
We should fix them all in a dedicated PR and put some lint steps in the build process. But it's not the purpose of this PR.

File: user/api.go (hunk 4 above)
Comment (CedricPerotto) [reply user/api.go]: Ok  I will push a fix on that.

File: Dockerfile (hunk 1 above)
Comment (CedricPerotto) [reply Dockerfile:13]: The image could be used in standalone for development purpose. and so on should be able to start properly.

**Review by yvon-dblg — APPROVED** [review]
LGTM

**Pull request merged**